### PR TITLE
Parse Java system properties in plugin.bat

### LINKED ISSUE #13616: plugin.bat error parsing Java options
When running `plugin.bat` on Windows, passing Java options via the command line results in the following error:

```
c:\elasticsearch\bin>plugin "-Des.plugins.staging=true" install analysis-icu
ERROR: unknown command [-Des.plugins.staging=true]. Use [-h] option to list available commands
```

Adding `-Des.plugins.staging=true` directly to the script [here](https://github.com/elastic/elasticsearch/blob/master/distribution/src/main/resources/bin/plugin.bat#L14) works.

This might be because `plugin.bat` is missing the equivalent of [this in the bash script.](https://github.com/elastic/elasticsearch/blob/master/distribution/src/main/resources/bin/plugin#L69-L83).

### PR DESCRIPTION
**Pull request by ywelsch**
Fixes #13616

### CODE REVIEW

File: distribution/src/main/resources/bin/plugin.bat [hunk 1]
```diff
@@ -9,9 +9,46 @@ for %%I in ("%SCRIPT_DIR%..") do set ES_HOME=%%~dpfI
...
 
+SET properties=
+SET args=
+
+:loop
+SET "current=%~1"
+SHIFT
+IF "x!current!" == "x" GOTO breakloop
+
+IF "!current:~0,2%!" == "-D" (
```
Comment (brwe) [inline distribution/src/main/resources/bin/plugin.bat]: why do we need the distinction of -D and -D*=* ? Can we not handle it like here https://github.com/elastic/elasticsearch/blob/master/distribution/src/main/resources/bin/elasticsearch.bat ? I don't actually know a lot of bat so I am really just curious.

Comment (ywelsch) [inline distribution/src/main/resources/bin/plugin.bat]: The idea is to model the same functionality as in the bash file:

https://github.com/elastic/elasticsearch/blob/master/distribution/src/main/resources/bin/plugin#L83-L90

This means that properties can be passed as either `-Dmy.prop true` or `-Dmy.prop=true`.


As for why the parsing is not done using `FOR /...` as in elasticsearch.bat, I could not get that approach to work with quoted parameters, for example `plugin "-Dmy.property=some value with spaces" install foo`.

Comment (brwe) [inline distribution/src/main/resources/bin/plugin.bat]: > This means that properties can be passed as either -Dmy.prop true or -Dmy.prop=true

ha! I did not even know we can do this! thanks for clarifying. But that means that it is different in `elasticsearch.bat` and also the bash elasticsearch script. I just checked, `-Dmy.prop true` does not work with them. I will open an issue for it.

Comment (rjernst) [inline distribution/src/main/resources/bin/plugin.bat]: That does not work with java. We should just stick with the way the jvm takes sysprops...
```
[14:04:26][~/Code/test]$ java -cp . Test
null
[14:04:50][~/Code/test]$ java -cp . -Dfoo bar Test
Error: Could not find or load main class bar
[14:05:03][~/Code/test]$ java -cp . -Dfoo=bar Test
bar
```

**Comment by gmarz** [comment]
LGTM- tested on my Windows laptop.  Thanks for fixing this @ywelsch

**Comment by clintongormley** [comment]
@ywelsch can we get this in?

**Pull request merged**